pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -628,6 +631,51 @@ type (
 	}
 )
 
+// Git groups all configuration options related to enabling GitOps on a
+// cluster and linking it to a Git repository.
+type Git struct {
+	Repo *Repo `json:"repo,omitempty"`
+	// +optional
+	Operator *Operator `json:"operator,omitempty"`
+	// +optional
+	Profiles []*Profile `json:"profiles,omitempty"` // one or many profiles to enable on this cluster
```

Comment:
Have we already implemented support for multiple profiles? I was hoping we could avoid that and keep it simple by only allowing a single profile per cluster. If we haven't implemented that yet, let's drop it from here.

Reply:
I believe this is now a requirement (cc @kalbir).

Besides, it seems legit to me, in order to mix and match simple profiles instead of having a combinatorial explosion of heavyweight profiles each with small differences (e.g. "_oh, but I prefer to use X than Y, let's fork and change just this bit, and keep the 50 other tools as is_").

It is kind of similar to "_composition_" v.s. "_inheritance_": either you compose what you need in your `ClusterConfig` or you have to come up with the perfect profile for everyone.

Reply:
Yes the intention was to have some simple predictable behaviour here. Not any smart composition but just something that we can communicate to people e.g. if you add a second profile it will overwrite files in the first.

Reply:
Note that even if the API has a list it doesn't mean we need to support it. It could be done just for the shake of "extendability" and forward compatibility. If we are unsure of the effects of adding more than one profile or we don't want to implement it just yet, we can always return an error when there is more than one item.